extensions/AzureCosmosDBMongoDB/AzureCosmosDBMongoDB/AzureCosmosDBMongoDBMemory.cs

```diff
@@ -0,0 +1,320 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading;
+using System.Threading.Tasks;
+using System.Runtime.CompilerServices;
+using Microsoft.Extensions.Logging;
+using Microsoft.KernelMemory.AI;
+using Microsoft.KernelMemory.Diagnostics;
+using Microsoft.KernelMemory.MemoryStorage;
+using MongoDB.Bson;
+using MongoDB.Driver;
+
+namespace Microsoft.KernelMemory.MemoryDb.AzureCosmosDBMongoDB;
+
+/// <summary>
+/// Azure Cosmos DB for MongoDB connector for Kernel Memory,
+/// Read more about Azure Cosmos DB for MongoDB and vector search here:
+/// https://learn.microsoft.com/en-us/azure/cosmos-db/mongodb/vcore/
+/// https://learn.microsoft.com/en-us/azure/cosmos-db/mongodb/vcore/vector-search
+/// </summary>
+public class AzureCosmosDBMongoDBMemory : IMemoryDb
+{
+    private readonly ITextEmbeddingGenerator _embeddingGenerator;
+    private readonly ILogger<AzureCosmosDBMongoDBMemory> _log;
+    private readonly AzureCosmosDBMongoDBConfig _config;
+    private readonly MongoClient _cosmosDBMongoClient;
+    private readonly IMongoDatabase _cosmosMongoDatabase;
+    private readonly IMongoCollection<AzureCosmosDBMongoDBMemoryRecord> _cosmosMongoCollection;
+    private readonly string _collectionName;
+
+    /// <summary>
+    /// Create a new instance
+    /// </summary>
+    /// <param name="config">Azure Cosmos DB for MongoDB configuration</param>
+    /// <param name="embeddingGenerator">Text embedding generator</param>
+    /// <param name="mongoClient">MongoClient for the MongoDB cluster</param>
+    /// <param name="databaseName">Database name for MongoDB</param>
+    /// <param name="collectionName">Collection name for MongoDB</param>
+    /// <param name="log">Application logger</param>
+    public AzureCosmosDBMongoDBMemory(
```

Comment:
.NET DI is unable to use this ctor and create instances of the class, because it requires `string` parameters to be resolved:
```
ystem.InvalidOperationException
Unable to resolve service for type 'System.String' while attempting to activate 'Microsoft.KernelMemory.MemoryDb.AzureCosmosDBMongoDB.AzureCosmosDBMongoDBMemory'.
```

You should remove the `collectionName` and use the `index` parameter received by each method below, and move the `databaseName` parameter into the config class.